### PR TITLE
Add category page dashboard first iteration

### PR DESCRIPTION
**Pull request by ritmatter**
Ready to be merged.

### CODE REVIEW

File: api/handlers/user.js [hunk 1]
```diff
@@ -154,10 +154,37 @@ var UserHandler = {
...
           return res.status(503).send(err);
         });
       },
+
+      // Get leading experts for a specific catery
+      getByCategory: {
+        investors: function(req, res) {
+          var high = req.params.order === 'high' ? -1 : 1
+          var category = req.params.category;
+          User.getInvestorsByMetricForCategory(high, category, req.params.datatype).then(
```
Comment (ritmatter) [inline api/handlers/user.js]: supadynamic

File: api/models/User.js [hunk 2]
```diff
@@ -254,114 +254,103 @@ UserSchema.statics.getLeadersByTimeStamp = function(high) {
     .exec();
 };
 
-// Find the top 10 leaders by investor reps
-UserSchema.statics.getLeadersByExpertReps = function(high) {
```
Comment (ritmatter) [inline api/models/User.js]: I'm not sure why, but it appears that a lot of this stuff was duplicated

File: api/models/User.js [hunk 3]
```diff
@@ -254,114 +254,103 @@ UserSchema.statics.getLeadersByTimeStamp = function(high) {
...
-  ]).exec();
-};
-
-// Get ranked user ids for a given expert category, decreasing
-// Ranking done by total dividends
-UserSchema.statics.findRankedInvestors = function(category) {
+// Find the top 10 investors by metric for a given category
+// Metrics available are timeStamp, average return, total dividends, and rank
+// Currently used for the Investor Dashboard on the Category Page
+// Returns truncated data for efficiency
```
Comment (ritmatter) [inline api/models/User.js]: This is entirely new, github is just getting confused

File: api/models/User.js [hunk 4]
```diff
@@ -254,114 +254,103 @@ UserSchema.statics.getLeadersByTimeStamp = function(high) {
...
   return this.aggregate([
     { $match: { "portfolio.category": category }},
     { $unwind: "$portfolio" },
     { $unwind: "$portfolio.investments" },
-    { $match: { "portfolio.category": category } },
-    { $group: { _id: "$_id", dividends: { $sum: "$portfolio.investments.dividend" } }},
-    { $sort: { "dividends": -1 }},
-    { $project: { "_id": 1 } },
+    { $match: { "portfolio.category": category }},
+    { $group:
```
Comment (ritmatter) [inline api/models/User.js]: Grouping requires fields to be inside an object. we put all the old stuff we want to keep inside the _id json block and then unpack it in the $project.

File: api/models/User.js [hunk 5]
```diff
@@ -254,114 +254,103 @@ UserSchema.statics.getLeadersByTimeStamp = function(high) {
...
+        dividends: 1
+      }
+    },
+    { $sort: sorter },
+    { $limit: 10 },
   ]).exec();
 };
 
-// Update the rank for a given investor and category
-UserSchema.statics.updateRank = function(userId, categoryName, rank, expert, cb) {
```
Comment (ritmatter) [inline api/models/User.js]: More apparently duplicated code

**Comment by krazemon** [comment]
LGTM

**Pull request merged**